website/blog/2023-06-12-multiple-same-components.md

```diff
@@ -0,0 +1,289 @@
+---
+title: Using multiple same components
+description: Explains how to use multiple same components inside
+slug: multiple-same-components
+authors: obradovic
+
+date: 2023-06-12
+tags: [eightshift, boilerplate, components, blocks]
+hide_table_of_contents: false
+---
+
+From time to time, you may have to create a block or a more complex component that is using the same component two or more times. In this blog post, we'll explain how it works in a bit more detail.
+<!--truncate-->
+
+An example of this use case is the Card component, which uses two heading components. You may get the general idea by just going through the code and trying to reverse-engineer it, but this example will give you a much better understanding of how it works and how to use it. First, let's cover some basics.
+
+## Manifest and attributes
+The way multiple same components work is by having a different key. Here is an example of the Card component:
+
+```json
+"components": {
+	"image": "image",
+	"intro": "heading",
+	"heading": "heading",
+	"paragraph": "paragraph",
+	"button": "button"
+},
+```
+As you can see, one heading component has the `intro` key, while the other one has the `heading` key. That way the **intro** heading component is being referred to as `intro` so there is no mixup with the attribute values between the two heading blocks. This can be seen when setting the default attributes:
+
+```json
+"cardIntroSize": {
+	"type": "string",
+	"default": "tiny"
+},
+"cardHeadingSize": {
+	"type": "string",
+	"default": "big"
+}
+```
+
+## The "props" Helper
+This method does all the heavy lifting for us. It replaces the default attribute names with the ones we provide. There is both the PHP and JS version of it.
+
+```php
+Components::render(
+	'heading',
+	Components::props('intro', $attributes, [
+		'selectorClass' => 'intro',
+		'blockClass' => $componentClass
+	])
+),
```

Comment:
```suggestion
Components::render('heading', Components::props('intro', $attributes, [
	'selectorClass' => 'intro',
	'blockClass' => $componentClass
]))
```
Just a suggestion to be more consistent with the props helper inside (first argument on the same line as the function call).

If you like how the existing one looks, feel free to ignore, but then just remove the comma at the end 😄